rust/processor/src/db/common/models/resources.rs

```diff
@@ -0,0 +1,212 @@
+    fungible_asset_models::v2_fungible_asset_utils::{
+        ConcurrentFungibleAssetBalance, ConcurrentFungibleAssetSupply, FungibleAssetMetadata,
+        FungibleAssetStore, FungibleAssetSupply,
+    },
+    object_models::v2_object_utils::{ObjectCore, Untransferable},
+    token_v2_models::v2_token_utils::{
+        AptosCollection, Collection, ConcurrentSupply, FixedSupply, PropertyMapModel,
+        TokenIdentifiers, TokenV2, UnlimitedSupply,
+    },
+};
+use anyhow::Result;
+use aptos_protos::transaction::v1::WriteResource;
+use const_format::formatcp;
+
+pub const COIN_ADDR: &str = "0x0000000000000000000000000000000000000000000000000000000000000001";
+pub const TOKEN_ADDR: &str = "0x0000000000000000000000000000000000000000000000000000000000000003";
+pub const TOKEN_V2_ADDR: &str =
+    "0x0000000000000000000000000000000000000000000000000000000000000004";
+
+pub const TYPE_FUNGIBLE_ASSET_SUPPLY: &str = formatcp!("{COIN_ADDR}::fungible_asset::Supply");
+pub const TYPE_CONCURRENT_FUNGIBLE_ASSET_SUPPLY: &str =
+    formatcp!("{COIN_ADDR}::fungible_asset::ConcurrentSupply");
+pub const TYPE_FUNGIBLE_ASSET_METADATA: &str = formatcp!("{COIN_ADDR}::fungible_asset::Metadata");
+pub const TYPE_FUNGIBLE_ASSET_STORE: &str = formatcp!("{COIN_ADDR}::fungible_asset::FungibleStore");
+pub const TYPE_CONCURRENT_FUNGIBLE_ASSET_BALANCE: &str =
+    formatcp!("{COIN_ADDR}::fungible_asset::ConcurrentFungibleBalance");
+
+pub const TYPE_OBJECT_CORE: &str = formatcp!("{COIN_ADDR}::object::ObjectCore");
+pub const TYPE_UNTRANSFERABLE: &str = formatcp!("{COIN_ADDR}::object::Untransferable");
+pub const TYPE_COLLECTION: &str = formatcp!("{TOKEN_V2_ADDR}::collection::Collection");
+pub const TYPE_CONCURRENT_SUPPLY: &str = formatcp!("{TOKEN_V2_ADDR}::collection::ConcurrentSupply");
+pub const TYPE_FIXED_SUPPLY: &str = formatcp!("{TOKEN_V2_ADDR}::collection::FixedSupply");
+pub const TYPE_UNLIMITED_SUPPLY: &str = formatcp!("{TOKEN_V2_ADDR}::collection::UnlimitedSupply");
+pub const TYPE_APOTS_COLLECTION: &str = formatcp!("{TOKEN_V2_ADDR}::aptos_token::AptosCollection");
+pub const TYPE_TOKEN_V2: &str = formatcp!("{TOKEN_V2_ADDR}::token::Token");
+pub const TYPE_TOKEN_IDENTIFIERS: &str = formatcp!("{TOKEN_V2_ADDR}::token::TokenIdentifiers");
+pub const TYPE_PROPERTY_MAP: &str = formatcp!("{TOKEN_V2_ADDR}::property_map::PropertyMap");
+
+pub trait Resource {
+    fn type_str() -> &'static str;
+}
+
+pub trait FromWriteResource<'a> {
+    fn from_write_resource(write_resource: &'a WriteResource) -> Result<Option<Self>>
+    where
+        Self: Sized;
+}
+
+impl<'a, T> FromWriteResource<'a> for T
+where
+    T: TryFrom<&'a WriteResource, Error = anyhow::Error> + Resource,
+{
+    fn from_write_resource(write_resource: &'a WriteResource) -> Result<Option<Self>> {
+        if MoveResource::get_outer_type_from_write_resource(write_resource) != Self::type_str() {
+            return Ok(None);
+        }
+        Ok(Some(write_resource.try_into()?))
```

Comment:
Can we log the txn_version and wsc_index in case this fails? That will help us debug errors in parsing logic